src/chains/kg_chain.py

```diff
@@ -70,32 +70,38 @@ def as_retrival_chain(self):
         )
 
     def as_generative_chain(self):
-        retrival_chain = self.as_retrival_chain()
+        retrival_chain = RunnableParallel(
```

Comment:
I think if we need to fix the retrival chain we should do it in the as_retrival_chain function .